internal/cache/controller_obj_cache.go

```diff
@@ -2269,6 +2269,19 @@ func (c *AviObjCache) AviObjOneVSCachePopulate(client *clients.AviClient, cloud
 						}
 					}
 				}
+				if vs["pool_group_ref"] != nil {
+					pgRef, ok := vs["pool_group_ref"].(string)
+					if ok {
+						pgUuid := ExtractUuid(pgRef, "poolgroup-.*.#")
+						pgName, foundpg := c.PgCache.AviCacheGetNameByUuid(pgUuid)
+						if foundpg {
+							pgKey := NamespaceName{Namespace: lib.GetTenant(), Name: pgName.(string)}
+							poolgroupKeys = append(poolgroupKeys, pgKey)
+							pgpoolKeys := c.AviPGPoolCachePopulate(client, cloud, pgName.(string))
```

Comment:
Since this method will be calledfrom layer3 during retries, should we always append pgKey to poolgroupKeys. Will this not lead to duplicate poolgroup entires being appended

Reply:
This method **AviObjOneVSCachePopulate** will refresh the cache for a specified vs every time it is invoked. I think the data will always be over-written.